Makefile

```diff
@@ -13,41 +7,67 @@ ifeq ($(TAGGED_VERSION),)
 endif
 VERSION ?= $(shell echo $(TAGGED_VERSION) | cut -c 2-)
 
-.PHONY: compare-deps
-compare-deps: Gopkg.lock GlooE-Gopkg.lock print-info
-	go run scripts/compare_dependencies.go Gopkg.lock GlooE-Gopkg.lock
+#----------------------------------------------------------------------------------
+# Retrieve GlooE build information
+#----------------------------------------------------------------------------------
+GLOOE_DIR := _glooe
+_ := $(shell mkdir -p $(GLOOE_DIR))
+
+.PHONY: get-glooe-info
+get-glooe-info: $(GLOOE_DIR)/Gopkg.lock $(GLOOE_DIR)/verify-plugins-linux-amd64 $(GLOOE_DIR)/build_env
 
-GlooE-Gopkg.lock:
-	curl -o GlooE-Gopkg.lock http://storage.googleapis.com/gloo-ee-dependencies/$(GLOOE_VERSION)/Gopkg.lock
+$(GLOOE_DIR)/Gopkg.lock:
+	curl -o $@ http://storage.googleapis.com/gloo-ee-dependencies/$(GLOOE_VERSION)/Gopkg.lock
+
+$(GLOOE_DIR)/verify-plugins-linux-amd64:
+	curl -o $@ http://storage.googleapis.com/gloo-ee-dependencies/$(GLOOE_VERSION)/verify-plugins-linux-amd64
+
+$(GLOOE_DIR)/build_env:
+	curl -o $@ http://storage.googleapis.com/gloo-ee-dependencies/$(GLOOE_VERSION)/build_env
 
-# TODO: remove
-.PHONY: print-info
-print-info:
-	@echo BUILD_ID: $(BUILD_ID)
-	@echo TAGGED_VERSION: $(TAGGED_VERSION)
-	@echo VERSION: $(VERSION)
-	@echo RELEASE: $(RELEASE)
 
 #----------------------------------------------------------------------------------
-# Build, test and publish example plugins
+# Compare dependencies against GlooE
+#----------------------------------------------------------------------------------
+
+.PHONY: compare-deps
+compare-deps: Gopkg.lock $(GLOOE_DIR)/Gopkg.lock
+	go run scripts/compare_dependencies.go Gopkg.lock $(GLOOE_DIR)/Gopkg.lock
+
+
+#----------------------------------------------------------------------------------
+# Build plugins
 #----------------------------------------------------------------------------------
 EXAMPLES_DIR := examples
 SOURCES := $(shell find . -name "*.go" | grep -v test)
 
-.PHONY: publish-examples
-publish-examples:
+define get_glooe_var
+$(shell grep $(1) $(GLOOE_DIR)/build_env | cut -d '=' -f 2-)
+endef
+
+.PHONY: build-plugins
+build-plugins: $(GLOOE_DIR)/build_env $(GLOOE_DIR)/verify-plugins-linux-amd64
+	docker build --no-cache -t quay.io/solo-io/ext-auth-plugins:$(VERSION) \
+		--build-arg GO_BUILD_IMAGE=$(call get_glooe_var,GO_BUILD_IMAGE) \
+		--build-arg GC_FLAGS=$(call get_glooe_var,GC_FLAGS) \
```

Comment:
looks like we're hard-coding all the possible values in `build_env` here. what if we wanted to add a new `FLAGS` build arg that isn't for GC, or we wanted to split out `GO_BUILD_IMAGE` into its constituent parts? Seems like we should just take all the env vars in the `build_env` we download and pass them through to the container

Reply:
Ah, the dockerfile using these variables is also in this repo, so I feel less strongly about this now

Reply:
LGTM 👍